src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -12884,7 +13523,197 @@ void Assembler::prefixq(Address adr, XMMRegister src) {
   }
 }
 
+void Assembler::prefixq_rex2(Address adr, XMMRegister src) {
+  if (src->encoding() >= 24) {
+    if (adr.base()->encoding() >= 24) {
+      if (adr.index()->encoding() >= 24) {
```

Comment:
The prefixq_rex2 can be simplified using the above get_base_or_dst_prefixq_bits, get_index_prefixq_bits, and get_src_prefixq_bits:
if (src->encoding() >= 8 || adr.base()->encoding() >= 8 || adr.index()->encoding()) {
  int bits = REXBIT_W; 
  bits |= get_base_or_dst_prefix_bits(adr.base()->encoding());
  bits |= get_index_prefix_bits(adr.index()->encoding());
  bits |= get_src_prefix_bits(src->encoding());
}
Do you think this would work or any issues with this approach?

Reply:
Done. Thanks.